BrainPortal/cbrain_plugins/cbrain-plugins-base/cbrain_task/boutiques_task/views/_show_params.html.erb

```diff
@@ -59,18 +71,27 @@
           Parameters
         </th>
       </tr>
-    <%- params.each do |param| -%>
-      <tr>
-        <td class="tsk-sw-name"><%= param.name %></td>
-        <td class="tsk-sw-val">
-          <% if param.type == 'Flag' %>
-            <%= red_if(@task.invoke_params[param.id].to_s =~ /^1$|^true$/, "No", "Yes", { :color1 => 'red', :color2 => 'green' }) %>
-          <% else %>
-            <%= format_param.(@task.invoke_params[param.id]) %>
-          <% end %>
-        </td>
-      </tr>
-    <%- end -%>
+      <%- params.each do |param| -%>
+        <%= render :partial => task_partial(:show_param), :locals => { :param => param, :format_param => format_param } %>
+      <%- end -%>
+    </tbody>
+  <%- end -%>
+  <%- unless params_with_groups.empty? -%>
+    <tbody class="tsk-sw-param">
+      <%- groups.each do |group| -%>
+        <%- members = group.members || [] -%>
+        <%- params_members = members - file_inputs_ids -%>
+        <%- next if params_members.empty? -%>
+        <tr>
+          <th colspan="2" class="tsk-sw-sub-hdr">
+            <%= group.name %>
+          </th>
+        </tr>
+        <%- params_members.each do |param_id| -%>
+          <%- param = params_with_groups.find { |p| p.id == param_id } -%>
```

Comment:
Why do you go through this complex process of creating params_with_groups wjhe you can simply find the param by id directly from the descriptor? `descriptor.input_by_id(param_id)`